test/unit/mdc-select/mdc-select-enhanced.test.js

```diff
@@ -0,0 +1,654 @@
+  component.selectedIndex = 1;
+  td.verify(mockFoundation.setSelectedIndex(1), {times: 1});
+});
+
+test('#set disabled calls foundation.updateDisabledStyle_', () => {
+  const hasMockFoundation = true;
+  const hasMockMenu = true;
+  const hasOutline = false;
+  const hasLabel = true;
+  const {component, mockFoundation} = setupTest(hasOutline, hasLabel, hasMockFoundation, hasMockMenu);
+  component.disabled = true;
+  td.verify(mockFoundation.setDisabled(true), {times: 1});
+});
+
+test('#initialSyncWithDOM sets the selected index if an option has the selected class', () => {
+  const fixture = bel`
+    <div class="mdc-select">
+      <div class="mdc-select__selected-text"></div>
+      <div class="mdc-select__menu mdc-menu mdc-menu-surface">
+      <ul class="mdc-list">
+        <li class="mdc-list-item"></li>
+        <li class="mdc-list-item mdc-list-item--selected" value="orange">          
+          Orange
+        </li>
+        <li class="mdc-list-item" value="apple">
+          Apple
+        </li> 
+      </ul>
+      </div>
+      <label class="mdc-floating-label">Pick a Food Group</label>
+      <div class="mdc-line-ripple"></div>
+    </div>
+  `;
+  const component = new MDCSelect(fixture, /* foundation */ undefined);
+  assert.equal(component.selectedIndex, 1);
+});
+
+test('#initialSyncWithDOM disables the select if the disabled class is found', () => {
+  const fixture = bel`
+    <div class="mdc-select mdc-select--disabled">
+      <div class="mdc-select__selected-text"></div>
+      <div class="mdc-select__menu mdc-menu mdc-menu-surface">
+      <ul class="mdc-list">
+        <li class="mdc-list-item mdc-list-item--selected"></li>
+        <li class="mdc-list-item" value="orange">          
+          Orange
+        </li>
+        <li class="mdc-list-item" value="apple">
+          Apple
+        </li> 
+      </ul>
+      </div>
+      <label class="mdc-floating-label">Pick a Food Group</label>
+      <div class="mdc-line-ripple"></div>
+    </div>
+  `;
+  const component = new MDCSelect(fixture);
+  assert.isTrue(component.disabled);
+});
+
+test('adapter#addClass adds a class to the root element', () => {
+  const {component, fixture} = setupTest();
+  component.getDefaultFoundation().adapter_.addClass('foo');
+  assert.isTrue(fixture.classList.contains('foo'));
+});
+
+test('adapter#removeClass removes a class from the root element', () => {
+  const {component, fixture} = setupTest();
+  fixture.classList.add('foo');
+  component.getDefaultFoundation().adapter_.removeClass('foo');
+  assert.isFalse(fixture.classList.contains('foo'));
+});
+
+test('adapter#hasClass returns true if a class exists on the root element', () => {
+  const {component, fixture} = setupTest();
+  fixture.classList.add('foo');
+  assert.isTrue(component.getDefaultFoundation().adapter_.hasClass('foo'));
+});
+
+test('adapter_.floatLabel does not throw error if label does not exist', () => {
```

Comment:
Please use `adapter#methodName` notation for all of these descriptions (it's a mixed bag from here down)